subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/internal/PropertyReportRenderer.java

```diff
@@ -27,6 +31,13 @@
      * @param value The value of the property
      */
     public void addProperty(String name, Object value) {
-        getTextOutput().formatln("%s: %s", name, value);
+        String strValue;
+        try {
+            strValue = String.valueOf(value);
+        } catch (Exception e) {
+            LOGGER.warn("Rendering of the property '{}' with value type '{}' failed with exception", name, value.getClass(), e);
```

Comment:
Is value always non-null here? It looks like it might be nullable, so value.getClass would throw a NPE.

Reply:
IDEA deceived me there, it was trusting the fact that the lack of the parameter annotation == NonNull. Adding `Nullable` removed the confusion so I added a null check. I'm still pretty sure that `valueOf` won't throw if the argument is null though.